lib/matplotlib/tests/test_text.py

```diff
@@ -444,8 +444,14 @@ def test_annotation_negative_fig_coords():
 
 
 def test_text_stale():
+    # A version of draw_all that draws even when interactive is off
+    def draw_all_when_not_interactive():
+        for f_mgr in plt._pylab_helpers.Gcf.get_all_fig_managers():
+            if f_mgr.canvas.figure.stale:
+                f_mgr.canvas.draw()
```

Comment:
Can this function be eliminated by switching to interactive mode?  If so, that seems like a better test.

Reply:
no, because we would never hit the "idle" case to actually let things draw, however I this can be done with direct call to `fig.canvas.draw()` (as we have the figures here!).